easybuild/tools/run.py

```diff
@@ -73,6 +74,105 @@
+        qa_patterns=None, qa_wait_patterns=None):
+    """
+    Run specified (interactive) shell command, and capture output + exit code.
+
+    :param fail_on_error: fail on non-zero exit code (enabled by default)
+    :param split_stderr: split of stderr from stdout output
+    :param stdin: input to be sent to stdin (nothing if set to None)
+    :param hidden: do not show command in terminal output (when using --trace, or with --extended-dry-run / -x)
+    :param in_dry_run: also run command in dry run mode
+    :param work_dir: working directory to run command in (current working directory if None)
+    :param shell: execute command through a shell (enabled by default)
+    :param output_file: collect command output in temporary output file
+    :param stream_output: stream command output to stdout
+    :param asynchronous: run command asynchronously
+    :param qa_patterns: list of 2-tuples with patterns for questions + corresponding answers
+    :param qa_wait_patterns: list of 2-tuples with patterns for non-questions
+                             and number of iterations to allow these patterns to match with end out command output
+    :return: Named tuple with:
+    - output: command output, stdout+stderr combined if split_stderr is disabled, only stdout otherwise
+    - exit_code: exit code of command (integer)
+    - stderr: stderr output if split_stderr is enabled, None otherwise
+    """
+
+    # temporarily raise a NotImplementedError until all options are implemented
+    if any((not fail_on_error, split_stderr, stdin, in_dry_run, work_dir, output_file, stream_output, asynchronous)):
+        raise NotImplementedError
+
+    if qa_patterns or qa_wait_patterns:
+        raise NotImplementedError
+
+    if isinstance(cmd, str):
+        cmd_msg = cmd.strip()
+    elif isinstance(cmd, list):
+        cmd_msg = ' '.join(cmd)
+    else:
+        raise EasyBuildError(f"Unknown command type ('{type(cmd)}'): {cmd}")
+
+    silent = build_option('silent')
+
+    if work_dir is None:
+        work_dir = os.getcwd()
+
+    # output file for command output (only used if output_file is enabled)
+    cmd_out_fp = None
+
+    # early exit in 'dry run' mode, after printing the command that would be run (unless 'hidden' is enabled)
+    if build_option('extended_dry_run'):
+        if not hidden:
+            msg = f"  running command \"%{cmd_msg}s\"\n"
+            msg += f"  (in %{work_dir})"
+            dry_run_msg(msg, silent=silent)
+
+        return RunResult(output='', exit_code=0, stderr=None)
+
+    start_time = datetime.now()
+    if not hidden:
+        cmd_trace_msg(cmd_msg, start_time, work_dir, stdin, cmd_out_fp)
+
+    _log.info(f"Running command '{cmd_msg}' in {work_dir}")
+    proc = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=shell)
+
+    # return output as a regular string (UTF-8 characters get stripped out)
+    output = proc.stdout.decode('ascii', 'ignore')
```

Comment:
now that we only support Python 3 we can just accept unicode strings by default

Reply:
OK, so we can use `utf-8` rather than `ascii` as encoding, but we still need to decode the `proc.stdout` bytes value to a string value, or we have to use byte sequence everywhere we play with the command output (which we really don't want to do).
In theory, the command output could contain non-UTF-8 characters (however unlikely), we still want to ignore those when decoding to avoid running into trouble.

Reply:
Switched to `utf-8` encoding in 1794660abf59855b22c59908dbf89ad761e03d0f